Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs

```diff
@@ -24,30 +36,220 @@ public class DafnyHoverHandler : HoverHandlerBase {
       };
     }
 
-    public async override Task<Hover?> Handle(HoverParams request, CancellationToken cancellationToken) {
+    public override async Task<Hover?> Handle(HoverParams request, CancellationToken cancellationToken) {
       logger.LogTrace("received hover request for {Document}", request.TextDocument);
       var document = await documents.GetDocumentAsync(request.TextDocument);
       if (document == null) {
         logger.LogWarning("the document {Document} is not loaded", request.TextDocument);
         return null;
       }
+      var diagnosticHoverContent = GetDiagnosticsHover(document, request.Position, out var areMethodStatistics);
       if (!document.SymbolTable.TryGetSymbolAt(request.Position, out var symbol)) {
         logger.LogDebug("no symbol was found at {Position} in {Document}", request.Position, request.TextDocument);
+      }
+
+      var symbolHoverContent = symbol != null ? CreateSymbolMarkdown(symbol, cancellationToken) : null;
+      if (diagnosticHoverContent == null && symbolHoverContent == null) {
         return null;
       }
-      return CreateHover(symbol, cancellationToken);
+
+      // If diagnostics are method diagnostics, we prioritize displaying the symbol information.
+      // This makes testing easier and less surprise for the user.
+      var hoverContent = areMethodStatistics && symbolHoverContent != null ? "" : (diagnosticHoverContent ?? "");
+      hoverContent = symbolHoverContent != null ? hoverContent + (hoverContent != "" ? "  \n" : "") + symbolHoverContent : hoverContent;
+      return CreateMarkdownHover(hoverContent);
+    }
+
+    class TupleComparer : IComparer<(int, int)> {
```

Comment:
Isn't this the default way of comparing tuples? https://stackoverflow.com/a/27781538/93197

Reply:
After I changes tuples to a record, I kept this class since the tuple comparison wouldn't work anymore, but thanks for this interesting finding !